### DISCUSSION TITLE
V5.1.1.0 / LR CC 11.0 / Camera Raw 14.0 / MacOS 12.0.1 /X-Touch mini -> Does not start

### DISCUSSION THREAD

**Discussion by RoSche2001** [rsjaffe/MIDI2LR#544] [Problems]
For the past weeks I could not get MIDI2LR to work. The problem has been described by others but solutions applied there did not work for me. I was waiting for updates from all packages involved to solve it magically, but it still fails.

-> The MIDI2LR window does not appear as usually at startup. LR module manager does not show any errors messages, module is shown as activated.
-> In MacOS in the past I had to enable Accessibility Access. Since I have the problem MIDI2LR does not show up there anymore. I tried to create an entry manually by copying the MIDI2LR app from the plugin package so that I could add it to accessability. Same for folder access (there was no entry before, I tried to create one manually). -> It did not help.
-> I just tried to procedure with your 2 reset .lua in the plugin folder. In my case the File ⇢ Plug-in Extras ⇢ MIDI2LR ⇢ Reset settings entry did not show up (there are several entries but no "Reset").
-> I can open several of the usual windows from the extras menu, also Start Application, but it does no good.

I could need some help in solving this please.
Cheers,
Robert

**Comment by rsjaffe** [comment]
Could you post the contents of the file  **~/Library/Logs/MIDI2LR/MIDI2LRinfo.txt**? Also, open up the MIDI2LR.lrplugin directory and try to start the MIDI2LR app. Does it start?

One way to approach fixing this is to delete the MIDI2LR.lrplugin directory and reinstall the application.

**Reply by RoSche2001** [reply]
I just did the following now:
- De-Installed MIDI2LR from LR, deleted the plugin folder (as several times over the past weeks)
- Rebooted, the LR open to double check that it does not show MIDI2LR, closed in library mode
- Installed MIDI2LR again
- MacOS asked for access to documents folder (but is still does not ask for accessibility nor does it show up there
- opened LR to double check -> still no success
- opened MIDI2LR plugin package and tried to open MIDI2LR there -> no success

MIDI2LRinfo.txt is this:
----------- LIGHTROOM -----------
Operating system 11.6.0.20G165 (x64)
Version 10.4
Language de
Preferences path /Users/roberto/Library/Preferences/Adobe/Lightroom
Application data path /Users/roberto/Library/Application Support/Adobe/Lightroom
----------- PLUGIN -----------
Version 4.3.1.11
Path /Users/roberto/Library/Application Support/Adobe/Lightroom/Modules/MIDI2LR.lrplugin
MIDI2LR log path /Users/roberto/Library/Logs/MIDI2LR
MIDI2LR data path /Users/roberto/Library/Application Support/MIDI2LR
----------- APP -----------
Application: System language de-DE.
Application: CPU Intel(R) Core(TM) i7-8700B CPU @ 3.20GHz.
Application: Application version 4.3.1.11.
Application: Application path /Users/roberto/Library/Application Support/Adobe/Lightroom/Modules/MIDI2LR.lrplugin/MIDI2LR.app.
Application: Profile directory /Users/roberto/Documents/_Lightroom Kataloge und Zubehör.
Application: Log file directory /Users/roberto/Library/Logs/MIDI2LR/.
Application: Settings file directory /Users/roberto/Library/Application Support/MIDI2LR/.19126



Entries in MIDI2LR.log are always the same (last three shown here):
**********************************************************

Log started: 15 Nov 2021 8:06:55pm

2021-11-15T20:06:55.384+01:00: MenuTrans.xml archive loaded from /Users/roberto/Library/Application Support/MIDI2LR/MenuTrans.xml.
2021-11-15T20:06:55.445+01:00: Exception Could not detect cereal root node - likely due to empty or invalid input void MIDI2LRApplication::LoadControlsModel().
2021-11-15T20:06:55.445+01:00: Exception Could not detect cereal root node - likely due to empty or invalid input virtual void MIDI2LRApplication::initialise(const juce::String &).
2021-11-15T20:06:55.445+01:00: Terminate called, exception Could not detect cereal root node - likely due to empty or invalid input.

**********************************************************

Log started: 15 Nov 2021 8:07:03pm

2021-11-15T20:07:03.022+01:00: MenuTrans.xml archive loaded from /Users/roberto/Library/Application Support/MIDI2LR/MenuTrans.xml.
2021-11-15T20:07:03.080+01:00: Exception Could not detect cereal root node - likely due to empty or invalid input void MIDI2LRApplication::LoadControlsModel().
2021-11-15T20:07:03.080+01:00: Exception Could not detect cereal root node - likely due to empty or invalid input virtual void MIDI2LRApplication::initialise(const juce::String &).
2021-11-15T20:07:03.081+01:00: Terminate called, exception Could not detect cereal root node - likely due to empty or invalid input.

**********************************************************

Log started: 15 Nov 2021 8:12:33pm

2021-11-15T20:12:33.234+01:00: MenuTrans.xml archive loaded from /Users/roberto/Library/Application Support/MIDI2LR/MenuTrans.xml.
2021-11-15T20:12:33.273+01:00: Exception Could not detect cereal root node - likely due to empty or invalid input void MIDI2LRApplication::LoadControlsModel().
2021-11-15T20:12:33.274+01:00: Exception Could not detect cereal root node - likely due to empty or invalid input virtual void MIDI2LRApplication::initialise(const juce::String &).
2021-11-15T20:12:33.274+01:00: Terminate called, exception Could not detect cereal root node - likely due to empty or invalid input.





Looks like either MacOS blocks the app or the app is missing something to go live. There is no additional virus scanner or similar enabled.

Many thanks for helping with this.

Robert

**Reply by RoSche2001** [reply]
One thought, not sure if this is relevant: Although I installed the package for version 5.1.1.0 the info file says version 4.3.1.11 (??)

**Comment by rsjaffe** [comment]
That's relevant. You have two different versions installed in two separate places.

Version 4.3.1.11 is installed at /Users/roberto/Library/Application Support/Adobe/Lightroom/Modules/MIDI2LR.lrplugin . Delete that. 
Find out where you installed 5.1.1.0. Delete that. Go into Lightroom, and delete MIDI2LR from the plugins manager. Get out of LR. 
Delete  /Users/roberto/Library/Application Support/MIDI2LR/.

Reinstall MIDI2LR in  /Users/roberto/Library/Application Support/Adobe/Lightroom/Modules/ .

**Reply by RoSche2001** [reply]
Deleted the plugin in the folder you spelled out.
When I tried to find out where the new version was installed, I opened the installer and it shows exactly the same folder as the one where I just deleted it.
Is there a slight chance that the V5.1.1.0 MacOS package had a wrong version included? I can't find another instance of this plugin on the whole computer. (What speaks against it is the fact that it stopped working a while ago...)
I keep looking and will also download the installer again.

**Comment by rsjaffe** [comment]
Try running it now and see if it works. If not, submit the MIDI2LRinfo.txt file again.

**Reply by RoSche2001** [reply]
Please note (not sure if relevant): I use the German language install.

After deleting the plugin, the installer, the log files I downloaded the installer again.
Did install again, double checked in the target folder, there it says version 5.1.1.0 for MIDI2LR program.
Opened LR, there the plugin was not activated, but showed the same version. Did activate manually. Then closed LR again.
The MIDI2LR.log file is now:
"**********************************************************

Log started: 15 Nov 2021 8:50:18pm

2021-11-15T20:50:18.728+01:00: MenuTrans.xml archive loaded from /Users/roberto/Library/Application Support/MIDI2LR/MenuTrans.xml.
2021-11-15T20:50:18.773+01:00: Exception Could not detect cereal root node - likely due to empty or invalid input void MIDI2LRApplication::LoadControlsModel().
2021-11-15T20:50:18.773+01:00: Exception Could not detect cereal root node - likely due to empty or invalid input virtual void MIDI2LRApplication::initialise(const juce::String &).
2021-11-15T20:50:18.774+01:00: Terminate called, exception Could not detect cereal root node - likely due to empty or invalid input.
"


There is no MIDI2LRinfo.txt file. The entries in MACOS folder access are gone as well.
Did try to open MIDI2LR from within the plugin package.
Did open/close LR several times, restarted the app from LR menu, but no change.

**Reply by RoSche2001** [reply]
Another observation: From within the LR module manager I can not delete MIDI2LR, the delete button is not activated (in contrary to other plugins there)

**Comment by rsjaffe** [comment]
delete  /Users/roberto/Library/Application Support/MIDI2LR/ and try again.

**Reply by RoSche2001** [reply]
YEEEEAAAAHHH!    It works!!!  😃 🎉   All is back to normal. 

That is wonderful -many thanks!!

**Reply by RoSche2001** [reply]
Have a wonderful week!!!!

**Comment by HGordon9** [comment]
![Screenshot 2021-11-19 at 13 24 50](https://user-images.githubusercontent.com/94693794/142635841-bb5bf5ac-c7fc-43da-9659-970f9142cee6.png)
MIDI2LR is not opening.  Showing installed and running and enabled but not opening  when I restart LR.  Tried un/reinstalling, tried older version but not luck.  Help would be apprectiated

**Reply by rsjaffe** [reply]
1. Is this the first time you've installed it or is this an update?
2. Please post the contents of _MIDI2LRinfo.txt_, found in _%AppData%\MIDI2LR_ (windows machines) or _~/Library/Logs/MIDI2LR_ (Macs).

**Comment by HGordon9** [comment]
Thanks for your reply.
This is the upgrade been working well before.
All I can find is the attached.  Not sure its what you mean.
Also this came up about the ports, no idea what it means.
        www.gordonphoto.co.uk

**Reply by rsjaffe** [reply]
Nothing is attached to your message

If something came up about the ports while doing a new install and Lightroom wasn't running when you were doing a new install then you have something on your machine (e.g., firewall) blocking or using those ports. You'll need to fix that.

Otherwise, I need to see the file I mentioned above to provide any more help.

**Comment by HGordon9** [comment]
This is what I sent but I don't think it is the one you mean.   I cannot
find the file you mentioned. ie *%AppData%\MIDI2LR. Can you suggest how I
can find it. Mac OS*

*Thanks*
        www.gordonphoto.co.uk

**Reply by rsjaffe** [reply]
For macs, look in ~/Library/Logs/MIDI2LR, not %AppData%\MIDI2LR.

**Comment by HGordon9** [comment]
Cannot find the Logs anywhere. I am sorry to say.

**Comment by HGordon9** [comment]
Managed to find this at last.
I hope it helps.
Thanks
        www.gordonphoto.co.uk

**Comment by HGordon9** [comment]
Earlier version of log
        www.gordonphoto.co.uk

**Reply by rsjaffe** [reply]
You're not including the log. Please post directly at https://github.com/rsjaffe/MIDI2LR/discussions/544 rather than by email.

**Comment by HGordon9** [comment]
Sorry, now I understand, Logs attached.

![Screenshot 2021-11-22 at 17 58 38](https://user-images.githubusercontent.com/94693794/142916969-8e90c9a0-0822-42a4-b413-7bcb9dfb739f.png)
![Screenshot 2021-11-22 at 17 50 00](https://user-images.githubusercontent.com/94693794/142916990-69e616d1-3c9d-45fc-bb10-c8620091b9a5.png)

**Reply by rsjaffe** [reply]
Try this. 
1. Close Lightroom.
2. Delete _/Users/howardgordon/Library/Application Support/MIDI2LR/MenuTrans.xml_. 
3. Restart Lightroom. 

Did that fix it?

**Comment by HGordon9** [comment]
I cannot find that. Library is not in that path. It goes User, Howard Gordon but no Library.  I've tried a global search it does not come up. Any suggestion how I find it?
![Screenshot 2021-11-22 at 19 45 02](https://user-images.githubusercontent.com/94693794/142925147-52a2f3ec-16ae-40f8-a5ef-9dd048b21c61.png)

**Reply by rsjaffe** [reply]
Use Google to find the answer.

**Comment by HGordon9** [comment]
Found it and deleted. Closed LR and restarted comp but still no joy.
![Screenshot 2021-11-22 at 20 29 31](https://user-images.githubusercontent.com/94693794/142930921-acd9f992-e763-4ab4-a7d5-78458f608e83.png)
![Screenshot 2021-11-22 at 20 32 17](https://user-images.githubusercontent.com/94693794/142931207-6310655e-0d6a-4f1c-882f-3c5ca219f562.png)

**Reply by rsjaffe** [reply]
There's something really messed up with your installation. Let's clear it all and reinstall.
1. Exit Lightroom
2. Delete the directory _/Users/howardgordon/Library/Application Support/MIDI2LR/_.
3. Delete directory _/Users/howardgordon/Library/Application Support/Adobe/Lightroom/Modules/MIDI2LR.lrplugin_.
4. Reinstall MIDI2LR.

**Comment by HGordon9** [comment]
It's working.  Thanks for your help,  Much appreciated.
